src/client/ua_client_connect.c

```diff
@@ -1197,8 +1219,12 @@ __Client_networkCallback(UA_ConnectionManager *cm, uintptr_t connectionId,
          * before sending the HEL message. */
         if(state == UA_CONNECTIONSTATE_OPENING)
             client->channel.state = UA_SECURECHANNELSTATE_CONNECTING;
-        else /* state == UA_CONNECTIONSTATE_ESTABLISHED */
-            client->channel.state = UA_SECURECHANNELSTATE_CONNECTED;
+        else {
+            if (client->channel.isReverseConnect)
```

Comment:
Test for `UA_SECURECHANNELSTATE_REVERSE_LISTENING` instead of the `isReverseConnect` flag.

Reply:
Done